### PR TITLE
Fixes VSTS Bug 829199: [Feedback] Document Outline doesn't work with New Editor

### PR DESCRIPTION
**Pull request by avodovnik**
Added support for the new extensibility approach via MEF in the outline view.

### CODE REVIEW

File: main/src/addins/CSharpBinding/MonoDevelop.CSharp.ClassOutline/CSharpOutlineTextEditorExtension.cs [hunk 1]
```diff
@@ -76,6 +78,18 @@ class CSharpOutlineTextEditorExtension : TextEditorExtension, IOutlinedDocument
 		bool disposed;
 		bool outlineReady;
 
+		Document providedAnalysisDocument;
```
Comment (Therzok) [inline main/src/addins/CSharpBinding/MonoDevelop.CSharp.ClassOutline/CSharpOutlineTextEditorExtension.cs]: Can we use a DocumentId here?

Comment (avodovnik) [reply main/src/addins/CSharpBinding/MonoDevelop.CSharp.ClassOutline/CSharpOutlineTextEditorExtension.cs]: I don't think so, as I need the document itself later down the line. https://github.com/mono/monodevelop/pull/7468/commits/fe5338ed32b80e781f30bf7924ec8a552ba77ed1#diff-3d2f21aec86f1047d4df1c8f3fe255d4R311

File: main/src/addins/CSharpBinding/MonoDevelop.CSharp.ClassOutline/CSharpOutlineTextEditorExtension.cs [hunk 2]
```diff
@@ -217,6 +231,18 @@ void JumpToDeclaration (bool focusEditor)
 			var o = outlineTreeStore.GetValue (IsSorting () ? outlineTreeModelSort.ConvertIterToChildIter (iter) : iter, 0);
 
 			var syntaxNode = o as SyntaxNode;
+
+			// if we can do it the "new" way, let's just do that ...
+			if (Editor == null) {
```
Comment (sandyarmstrong) [inline main/src/addins/CSharpBinding/MonoDevelop.CSharp.ClassOutline/CSharpOutlineTextEditorExtension.cs]: I think this check should maybe be `if (providedAnalysisDocument != null)`.

File: main/src/addins/CSharpBinding/MonoDevelop.CSharp.ClassOutline/CSharpOutlineTextEditorExtension.cs [hunk 3]
```diff
@@ -282,7 +308,7 @@ void RemoveRefillOutlineStoreTimeout ()
 		uint refillOutlineStoreId;
 		async void UpdateDocumentOutline (object sender, EventArgs args)
 		{
-			var analysisDocument = DocumentContext.AnalysisDocument;
+			var analysisDocument = DocumentContext?.AnalysisDocument ?? this.providedAnalysisDocument;
```
Comment (sandyarmstrong) [inline main/src/addins/CSharpBinding/MonoDevelop.CSharp.ClassOutline/CSharpOutlineTextEditorExtension.cs]: Should we reverse this? `providedAnalysisDocument` would be preferred, right?

File: main/src/addins/CSharpBinding/MonoDevelop.CSharp.ClassOutline/CSharpOutlineTextEditorExtension.cs [hunk 4]
```diff
@@ -44,6 +44,8 @@
 using MonoDevelop.Projects;
 using MonoDevelop.Ide.Editor.Extension;
 using MonoDevelop.Ide.Editor;
+using Microsoft.VisualStudio.Text.Editor;
+using MonoDevelop.TextEditor;
```
Comment (sandyarmstrong) [inline main/src/addins/CSharpBinding/MonoDevelop.CSharp.ClassOutline/CSharpOutlineTextEditorExtension.cs]: Let's sort these better.

File: main/src/addins/CSharpBinding/MonoDevelop.CSharp.ClassOutline/CSharpOutlineTextEditorExtension.cs (hunk 3 above)
Comment (avodovnik) [reply main/src/addins/CSharpBinding/MonoDevelop.CSharp.ClassOutline/CSharpOutlineTextEditorExtension.cs]: Yeah, good point.

File: main/src/addins/CSharpBinding/MonoDevelop.CSharp.ClassOutline/CSharpOutlineTextEditorExtension.cs (hunk 4 above)
Comment (avodovnik) [reply main/src/addins/CSharpBinding/MonoDevelop.CSharp.ClassOutline/CSharpOutlineTextEditorExtension.cs]: Turns out the two you highlighted weren't even used, so I cleared them out and sorted the rest.

**Comment by abock** [comment]
@monojenkins backport release-8.1

**Pull request merged**